filters/auth/grantlogout.go

```diff
@@ -0,0 +1,202 @@
+//
+// grantLogout filter
+//
+// Used for revoking access/refresh tokens generated by the oauthGrant() filter.
+//
+
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	GrantLogoutName           = "grantLogout"
+	revokeTokenKey            = "token"
+	revokeTokenTypeKey        = "token_type_hint"
+	refreshTokenType          = "refresh_token"
+	accessTokenType           = "access_token"
+	unsupportedTokenTypeError = "unsupported_token_type"
+)
+
+type grantLogoutSpec struct {
+	config OAuthConfig
+}
+
+type grantLogoutFilter struct {
+	config OAuthConfig
+}
+
+type revokeErrorResponse struct {
+	Error            string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+}
+
+type basicAuthCredentials struct {
+	clientId     string
+	clientSecret string
+}
+
+func (*grantLogoutSpec) Name() string { return GrantLogoutName }
+
+func (s *grantLogoutSpec) CreateFilter([]interface{}) (filters.Filter, error) {
+	return &grantLogoutFilter{
+		config: s.config,
+	}, nil
+}
+
+func (f *grantLogoutFilter) getBasicAuthCredentials() (*basicAuthCredentials, error) {
+	clientID := f.config.GetClientID()
+	if clientID == "" {
+		return nil, errors.New("failed to create token revoke auth header: no client ID")
+	}
+
+	clientSecret := f.config.GetClientSecret()
+	if clientSecret == "" {
+		return nil, errors.New("failed to create token revoke auth header: no client secret")
+	}
+
+	return &basicAuthCredentials{
```

Comment:
Maybe just `return clientID, clientSecret, nil`?

Reply:
For some reason I considered this to not be go-y. But I like it better this way. Fixed.